### PR TITLE
EASY-1291 upgrade project

### PR DESCRIPTION
**Pull request by rvanheest**
fixes EASY-1291

#### When applied it will
* upgrade the project to the newest standards

@DANS-KNAW/easy for review

### CODE REVIEW

**Comment by rvanheest** [comment]
@janvanmansum done the whole list, except for these two:

1. easy-dtap-role bijgewerkt. Eventuele losse ansible-role verwijderd. Readme bijwerken.
2. Service opstarten vanuit Command-klasse met run-service subcommando (indien van project met daemon). Geen main method, maar gebruik van scala's App klasse.

I'm not sure about (1), but I leave it as a reminder that there might be some changes to be done in easy-dtap that I'm not certain about.
Regarding (2), I didn't do that one yet given the state of the code. Let me know if you want it for this project.

File: debug-init-env.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+touch ${DATADIR}/easy-archive-bag.log
+
+echo "Creating staging area in $DATADIR..."
+mkdir -p ${DATADIR}/easy-archive-bag-staging
+
+echo "Copy bags to $DATADIR..."
+mkdir ${DATADIR}/bags
+cp -r src/test/resources/bags/basic-sequence-unpruned ${DATADIR}/bags/basic-sequence-unpruned
+
+echo "OK"
```
Comment (jo-pol) [inline debug-init-env.sh]: An OK might be misleading if the mkdir's or copy's fail, or am I missing some magic trick that aborts the script on an error? These checks are also missing in the [archetype](https://github.com/DANS-KNAW/easy-module-archetype/blob/master/src/main/resources/archetype-resources/debug-init-env.sh).

File: src/main/ansible/opt.sh [hunk 2]
```diff
@@ -16,11 +16,5 @@
 #
 
 
-ARGS=$@
-APPHOME=home
-. apphome.sh
+export PATH=$PATH:/opt/bin
```
Comment (jo-pol) [inline src/main/ansible/opt.sh]: Vlag dekt de lading niet: A script called run that doesn't run anything?

File: src/main/assembly/dist/cfg/application.properties [hunk 3]
```diff
@@ -1,5 +1,5 @@
-default.storage-service-url={{ easy_archive_bag_default_storage_service_url }}
-default.storage-service-username={{ easy_archive_bag_default_storage_service_username }}
-default.storage-service-password={{ easy_archive_bag_default_storage_service_password }}
-tempdir={{ easy_archive_bag_tempdir }}
-bag-index.uri=http://localhost:20120/
\ No newline at end of file
+default.storage-service-url=http://localhost:20110/
+default.storage-service-username=
+default.storage-service-password=
```
Comment (jo-pol) [inline src/main/assembly/dist/cfg/application.properties]: Are these values set somehow when creating DEASY?

File: src/main/rpm/2-post-install.sh [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+PHASE="POST-INSTALL"
+ARCHIVE_STAGING_DIR=/var/opt/dans.knaw.nl/tmp/easy-archive-bag-staging
+
+echo "$PHASE: START (Number of current installations: $NUMBER_OF_INSTALLATIONS)"
+
+if [ ! -d ${ARCHIVE_STAGING_DIR} ]; then
+    echo -n "Creating archive staging directory..."
+    mkdir -p ${ARCHIVE_STAGING_DIR}
+    chmod 777 ${ARCHIVE_STAGING_DIR}
+    echo "OK"
```
Comment (jo-pol) [inline src/main/rpm/2-post-install.sh]: See previous comment on misleading OK.

File: src/main/scala/nl.knaw.dans.easy.archivebag/BagitFacadeComponent.scala [hunk 5]
```diff
@@ -61,5 +73,16 @@ trait Bagit5FacadeComponent extends BagFacadeComponent {
       case cause: InvalidBagitFileFormatException => Failure(BagReaderException(bagDir, cause))
       case NonFatal(cause) => Failure(NotABagDirException(bagDir, cause))
     }
+
+    // TODO: canditate for easy-bagit-lib
+    private def getIsVersionOfFromUri(uri: URI): Try[UUID] = {
```
Comment (jo-pol) [inline src/main/scala/nl.knaw.dans.easy.archivebag/BagitFacadeComponent.scala]: unlogical or at least difficult to read grammar in the name

File: src/main/scala/nl.knaw.dans.easy.archivebag/Configuration.scala [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+import scala.io.Source
+
+case class Configuration(version: String, properties: PropertiesConfiguration)
+
+object Configuration {
+
+  def apply(): Configuration = {
+    val home = Paths.get(System.getProperty("app.home"))
+    val cfgPath = Seq(Paths.get(s"/etc/opt/dans.knaw.nl/easy-archive-bag/"), home.resolve("cfg"))
```
Comment (jo-pol) [inline src/main/scala/nl.knaw.dans.easy.archivebag/Configuration.scala]: A hard coded path? Ah, extract var `defaultPath` would not require me to read twice what happens. To be consistent also extract `home.resolve("cfg")` into `configuredPath`.

File: src/main/scala/nl.knaw.dans.easy.archivebag/EasyArchiveBag.scala [hunk 7]
```diff
@@ -106,7 +83,7 @@ object EasyArchiveBag extends Bagit5FacadeComponent with DebugEnhancedLogging {
       get.addHeader("Accept", "text/plain;charset=utf-8")
       val sw = new StringWriter()
       val response = http.execute(get)
-      if(response.getStatusLine.getStatusCode != 200) throw new IllegalStateException(s"Error retrieving bag-sequence for bag: $bagId")
+      if (response.getStatusLine.getStatusCode != 200) throw new IllegalStateException(s"Error retrieving bag-sequence for bag: $bagId")
```
Comment (jo-pol) [inline src/main/scala/nl.knaw.dans.easy.archivebag/EasyArchiveBag.scala]: Please meaningful constant names for 200 and 201.

File: src/test/resources/bags/basic-sequence-unpruned/README.txt [hunk 8]
```diff
@@ -0,0 +1,45 @@
+This is a sequence of three bags. First a is added, then b and then c.
```
Comment (jo-pol) [inline src/test/resources/bags/basic-sequence-unpruned/README.txt]: `a`: article or enumeration? It slows down reading.

File: src/main/ansible/opt.sh (hunk 2 above)
Comment (janvanmansum) [inline src/main/ansible/opt.sh]: The script is called `opt.sh`, so what do you mean?

File: src/test/resources/bags/basic-sequence-unpruned/README.txt (hunk 8 above)
Comment (janvanmansum) [inline src/test/resources/bags/basic-sequence-unpruned/README.txt]: @jo-pol : I am going to merge this one. Please, do a PR for the issues you raised above.

**Pull request merged**